ibtk/include/ibtk/HierarchyAveragedDataManager.h

```diff
@@ -0,0 +1,251 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2022 - 2022 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_HierarchyAveragedDataManager
+#define included_IBTK_HierarchyAveragedDataManager
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/SnapshotCache.h>
+#include <ibtk/ibtk_utilities.h>
+
+#include "CellVariable.h"
+#include "HierarchySideDataOpsReal.h"
+#include "IntVector.h"
+#include "SideVariable.h"
+#include "VisItDataWriter.h"
+#include "tbox/Database.h"
+#include "tbox/Pointer.h"
+
+#include <string>
+#include <vector>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+/*!
+ * \brief Class HierarchyAveragedDataManager provides a method of determining and storing average fields over periodic
+ * intervals.
+ */
+class HierarchyAveragedDataManager
+{
+public:
+    /*!
+     * The constructor for class HierarchyAveragedDataManager sets some default values and determines data centering.
```

Comment:
We should list these out, and also explain what should be in the input database.